File: ai-test-platform/frontend/src/types/enums.ts
/**
 * 枚举辅助函数和映射
 */

import {
  TestCaseStatus,
  TestCasePriority,
  DataType,
  ExpectedBehavior,
  HealingLevel,
  RiskLevel
} from './core';

// ==================== 数据类型 ====================

export const DataTypeLabels: Record<DataType, string> = {
  [DataType.VALID]: '正常数据',
  [DataType.BOUNDARY]: '边界数据',
  [DataType.INVALID]: '异常数据',
  [DataType.NULL]: '空值',
  [DataType.EMPTY]: '空字符串'
};

export const DataTypeColors: Record<DataType, string> = {
  [DataType.VALID]: 'bg-green-100 text-green-700',
  [DataType.BOUNDARY]: 'bg-yellow-100 text-yellow-700',
  [DataType.INVALID]: 'bg-red-100 text-red-700',
  [DataType.NULL]: 'bg-gray-100 text-gray-700',
  [DataType.EMPTY]: 'bg-gray-100 text-gray-700'
};

// ==================== 预期行为 ====================

export const ExpectedBehaviorLabels: Record<ExpectedBehavior, string> = {
  [ExpectedBehavior.SUCCESS]: '成功响应',
  [ExpectedBehavior.CLIENT_ERROR]: '客户端错误',
  [ExpectedBehavior.SERVER_ERROR]: '服务器错误'
};

export const ExpectedBehaviorColors: Record<ExpectedBehavior, string> = {
  [ExpectedBehavior.SUCCESS]: 'bg-green-100 text-green-700',
  [ExpectedBehavior.CLIENT_ERROR]: 'bg-orange-100 text-orange-700',
  [ExpectedBehavior.SERVER_ERROR]: 'bg-red-100 text-red-700'
};

// ==================== 优先级 ====================

export const PriorityLabels: Record<TestCasePriority, string> = {
  [TestCasePriority.CRITICAL]: 'P0',
  [TestCasePriority.HIGH]: 'P1',
  [TestCasePriority.MEDIUM]: 'P2',
  [TestCasePriority.LOW]: 'P3'
};

export const PriorityColors: Record<TestCasePriority, string> = {
  [TestCasePriority.CRITICAL]: 'bg-red-100 text-red-700',
  [TestCasePriority.HIGH]: 'bg-orange-100 text-orange-700',
  [TestCasePriority.MEDIUM]: 'bg-yellow-100 text-yellow-700',
  [TestCasePriority.LOW]: 'bg-gray-100 text-gray-700'
};

// ==================== 状态 ====================

export const StatusLabels: Record<TestCaseStatus, string> = {
  [TestCaseStatus.PENDING]: '待执行',
  [TestCaseStatus.RUNNING]: '执行中',
  [TestCaseStatus.PASSED]: '通过',
  [TestCaseStatus.FAILED]: '失败',
  [TestCaseStatus.SKIPPED]: '跳过',
  [TestCaseStatus.BLOCKED]: '阻塞'
};

export const StatusColors: Record<TestCaseStatus, string> = {
  [TestCaseStatus.PENDING]: 'bg-gray-100 text-gray-700',
  [TestCaseStatus.RUNNING]: 'bg-blue-100 text-blue-700',
  [TestCaseStatus.PASSED]: 'bg-green-100 text-green-700',
  [TestCaseStatus.FAILED]: 'bg-red-100 text-red-700',
  [TestCaseStatus.SKIPPED]: 'bg-yellow-100 text-yellow-700',
  [TestCaseStatus.BLOCKED]: 'bg-purple-100 text-purple-700'
};

// ==================== 修复级别 ====================

export const HealingLevelLabels: Record<HealingLevel, string> = {
  [HealingLevel.L1_RETRY]: 'L1-重试',
  [HealingLevel.L2_DATA]: 'L2-数据重建',
  [HealingLevel.L3_TOLERANCE]: 'L3-容错',
  [HealingLevel.L4_MANUAL]: 'L4-人工'
};

export const HealingLevelColors: Record<HealingLevel, string> = {
  [HealingLevel.L1_RETRY]: 'bg-blue-100 text-blue-700',
  [HealingLevel.L2_DATA]: 'bg-green-100 text-green-700',
  [HealingLevel.L3_TOLERANCE]: 'bg-yellow-100 text-yellow-700',
  [HealingLevel.L4_MANUAL]: 'bg-red-100 text-red-700'
};

// ==================== 风险等级 ====================

export const RiskLevelLabels: Record<RiskLevel, string> = {
  [RiskLevel.CRITICAL]: '严重',
  [RiskLevel.HIGH]: '高',
  [RiskLevel.MEDIUM]: '中',
  [RiskLevel.LOW]: '低'
};

export const RiskLevelColors: Record<RiskLevel, string> = {
  [RiskLevel.CRITICAL]: 'bg-red-100 text-red-700',
  [RiskLevel.HIGH]: 'bg-orange-100 text-orange-700',
  [RiskLevel.MEDIUM]: 'bg-yellow-100 text-yellow-700',
  [RiskLevel.LOW]: 'bg-gray-100 text-gray-700'
};

File: core/enums.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
核心枚举定义
"""

from enum import Enum

# ==================== 测试用例相关 ====================

class TestCaseStatus(Enum):
    """测试用例状态"""
    PENDING = "pending"
    RUNNING = "running"
    PASSED = "passed"
    FAILED = "failed"
    SKIPPED = "skipped"
    BLOCKED = "blocked"

class TestCasePriority(Enum):
    """测试用例优先级"""
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"

# ==================== 数据类型相关 ====================

class DataType(Enum):
    """测试数据类型"""
    VALID = "valid"           # 正常数据
    BOUNDARY = "boundary"     # 边界数据
    INVALID = "invalid"       # 异常数据
    NULL = "null"             # 空值
    EMPTY = "empty"           # 空字符串

class ExpectedBehavior(Enum):
    """预期行为"""
    SUCCESS = "success"              # 成功响应 (200-299)
    CLIENT_ERROR = "client_error"    # 客户端错误 (400-499)
    SERVER_ERROR = "server_error"    # 服务器错误 (500-599)

# ==================== 断言相关 ====================

class AssertionOperator(Enum):
    """断言操作符"""
    EQUALS = "equals"
    NOT_EQUALS = "not_equals"
    CONTAINS = "contains"
    NOT_CONTAINS = "not_contains"
    GREATER_THAN = "greater_than"
    LESS_THAN = "less_than"
    GREATER_THAN_OR_EQUAL = "greater_than_or_equal"
    LESS_THAN_OR_EQUAL = "less_than_or_equal"
    IN = "in"
    NOT_IN = "not_in"
    MATCHES = "matches"
    EXISTS = "exists"
    NOT_EXISTS = "not_exists"
    IS_TYPE = "is_type"

# ==================== 修复相关 ====================

class HealingLevel(Enum):
    """修复层级"""
    L1_RETRY = "L1"          # 环境问题（重试）
    L2_DATA = "L2"           # 数据问题（重建数据）
    L3_TOLERANCE = "L3"      # 不稳定（容错）
    L4_MANUAL = "L4"         # 断言失败（人工）

# ==================== 测试类型相关 ====================

class RiskLevel(Enum):
    """风险等级"""
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"

class TestType(Enum):
    """测试类型"""
    FUNCTIONAL = "functional"      # 功能测试
    PERFORMANCE = "performance"    # 性能测试
    SECURITY = "security"          # 安全测试
    COMPATIBILITY = "compatibility" # 兼容性测试
    BOUNDARY = "boundary"          # 边界测试
    EXCEPTION = "exception"        # 异常测试

# ==================== 执行相关 ====================

class ExecutionMode(Enum):
    """执行模式"""
    SEQUENTIAL = "sequential"      # 顺序执行
    PARALLEL = "parallel"          # 并行执行
    DISTRIBUTED = "distributed"    # 分布式执行

# ==================== 报告相关 ====================

class ReportFormat(Enum):
    """报告格式"""
    JSON = "json"
    HTML = "html"
    TXT = "txt"
    PDF = "pdf"
    EXCEL = "excel"
